Lib/pathlib.py

```diff
@@ -680,22 +686,26 @@ def is_reserved(self):
         name = self._tail[-1].partition('.')[0].partition(':')[0].rstrip(' ')
         return name.upper() in _WIN_RESERVED_NAMES
 
-    def match(self, path_pattern):
+    def match(self, path_pattern, *, case_sensitive=None):
         """
         Return True if this path matches the given pattern.
         """
+        if case_sensitive is None:
+            case_sensitive = _is_case_sensitive(self._flavour)
+        flags = re.NOFLAG if case_sensitive else re.IGNORECASE
```

Comment:
As this is quite closely tied to the `re` pattern compilation, I think this line should be moved into `_compile_pattern`